source/AI.cpp

```diff
@@ -3292,20 +3292,40 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Board target");
+
+			auto strategy = [&]() -> function<double(const shared_ptr<Ship> &)> {
+				Point current = ship.Position();
+				if (!distancePriority)
+				{
+					double agility = ship.Acceleration() * ship.TurnRate();
+					return [agility, this, &ship, current](const shared_ptr<Ship> &other) -> double {
+						double cost = this->Has(ship, other, ShipEvent::SCAN_OUTFITS) ? 
+							other.get()->Cost() : (other.get()->ChassisCost() * 2.);
+						return -agility * 2. * (cost * cost) / current.DistanceSquared(other.get()->Position());
+					};
+				}
+				// Default to distance-based strategy.
+				return [current](const shared_ptr<Ship> &other) -> double {
+					return current.DistanceSquared(other.get()->Position());
+				};
+			};
+
 			for(const shared_ptr<Ship> &other : ships)
 				if(CanBoard(ship, *other))
 				{
 					if(shift && !other->IsYours())
 						continue;
 
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double b = strategy()(other);
```

Comment:
You're recalculating which strategy gets used on every iteration.

```suggestion
					double b = strategy(other);
```

And above

```cpp
auto strategy = []
    /* ...  */
  }();
// ^
```

like before, but without the argument.